diff --git a/src/ifs/wl_surface/zwlr_layer_surface_v1.rs b/src/ifs/wl_surface/zwlr_layer_surface_v1.rs
index 3186a1db6..ca08d42f9 100644
--- a/src/ifs/wl_surface/zwlr_layer_surface_v1.rs
+++ b/src/ifs/wl_surface/zwlr_layer_surface_v1.rs
@@ -522,8 +522,11 @@ impl ZwlrLayerSurfaceV1 {
          let extents = self.surface.extents.get();
          let (width, height) = extents.size();
          let mut anchor = self.anchor.get();
-        if anchor == 0 {
-            anchor = LEFT | RIGHT | TOP | BOTTOM;
+        if anchor & (LEFT | RIGHT) == 0 {
+            anchor |= LEFT | RIGHT;
+        }
+        if anchor & (TOP | BOTTOM) == 0 {
+            anchor |= TOP | BOTTOM;
          }
          let (mt, mr, mb, ml) = self.margin.get();
          let ons = &output.node_state[LiveTL];
